real path of input file

Hello!

I have a question about how put the realname of input file ("for example afl allow put -f" )

I'm trying to do something like this ( on windows 10 with de dumb mode)

manul.py -i C:\Users\test\Downloads\manul-master\in -o C:\Users\test\Downloads\manul-master\out  -n 1  -s  "mspaint.exe @@" 

but in the process montitor show "C:\Users\test\Downloads\manul-master\out\0\mutations\.cur_input"

regards

C:\Users\test\Downloads\manul-master>manul.py -i in -o out -n 5 "C:\Program
Files (x86)\Kaspersky Lab\Kaspersky Internet Security 20.0\avp.exe SCAN @@"
[ERROR] Unable to find binary C:Program (required)

El jue., 25 jun. 2020 a las 15:06, Juan Pablo Lopez Yacubian (<
jplopezy@gmail.com>) escribió:

>
> Yes sure, i 'm trying to fuzz Kaspersky so the full path is
>
> "C:\Program Files (x86)\Kaspersky Lab\Kaspersky Internet Security
> 20.0\avp.exe"
>
> but i'm getting an error because fail to parse this path
>
>
> [image: imagen.png]
> regards
>
> El mié., 24 jun. 2020 a las 18:56, Maksim Shudrak (<
> notifications@github.com>) escribió:
>
>> Could you please provide a full example of what you are trying to execute?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mxmssh/manul/issues/66#issuecomment-649094222>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABVQJUS55SIOMXZSBW3GTRTRYJZAXANCNFSM4OESXMRA>
>> .
>>
>

--- a/manul.config
+++ b/manul.config
@@ -97,3 +97,6 @@ user_signals = 6
 
 # Enable AFL's forkserver fuzzing mode (only available on Linux, experimental)
 #forkserver_on = True
+
+# Skip binary check for available instrumentation and path correctness
+skip_binary_check = False

--- a/manul.py
+++ b/manul.py
@@ -1387,11 +1387,12 @@ def parse_args():
     parser.add_argument("--disable_volatile_bytes", default = None, action = 'store_true', help = argparse.SUPPRESS)
     parser.add_argument("--stop_after_nseconds", default = 0.0, type=int, help = argparse.SUPPRESS)
     parser.add_argument("--forkserver_on", default = False, action = 'store_true', help = argparse.SUPPRESS)
-
+    parser.add_argument("--skip_binary_check", default = False, action = 'store_true', help = argparse.SUPPRESS)
 
     parser.add_argument('target_binary', nargs='*', help="The target binary and options to be executed (quotes needed e.g. \"target -png @@\")")
 
     args = parser.parse_args()
+
     additional_args = parse_config(args.config)
     # A little hack here. We actually adding commands from config to cmd string and then parse it all together.
     final_cmd_to_parse = "%s %s" % (" ".join(sys.argv[1:-1]), additional_args)
@@ -1449,9 +1450,10 @@ if __name__ == "__main__":
     if args.dbi is not None:
         dbi_setup = configure_dbi(args, target_binary, args.debug)
 
-    check_binary(target_binary)  # check if our binary exists and is actually instrumented
+    if not args.skip_binary_check:
+        check_binary(target_binary)  # check if our binary exists and is actually instrumented
 
-    if not args.simple_mode and args.dbi is None and not check_instrumentation(target_binary):
+    if not args.simple_mode and args.dbi is None and not args.skip_binary_check and not check_instrumentation(target_binary):
         ERROR("Failed to find afl's instrumentation in the target binary, try to recompile or run manul in dumb mode")
 
     if not os.path.isdir(args.input):

